cognite/client/_api/files.py

```diff
@@ -748,15 +786,14 @@ def _process_file_download(
         directory: Path,
         identifier: dict[str, int | str],
         id_to_metadata: dict[str | int, FileMetadata],
+        file_path: Path,
     ) -> None:
-        id = IdentifierSequence.unwrap_identifier(identifier)
-        file_metadata = id_to_metadata[id]
-        file_path = (directory / cast(str, file_metadata.name)).resolve()
-        file_is_in_download_directory = directory.resolve() in file_path.parents
+        file_path_absolute = file_path.resolve()
+        file_is_in_download_directory = directory.resolve() in file_path_absolute.parents
```

Comment:
Probably not worth the trouble here, but `PurePath.is_relative_to` could maybe have been used.

Reply:
Nevermind, its a 3.9 feature 😅